### ISSUE TITLE
Passing an empty array or null into ElasticAdapter\Search\SearchRequest

### ISSUE THREAD

**Issue by davisriska** [babenkoivan/elastic-adapter#23]
I'm currently using `babenkoivan/elastic-scout-driver-plus` to provide search ability for my Laravel models. I went through the code to figure out how I could skip the query and just get aggregations and some default/random models. 
I figured out that the `babenkoivan/elastic-scout-driver-plus` is using ElasticAdapter\Search\SearchRequest. To be able to skip the query part I implemented a workaround like so 
composer.json
```json
...
"autoload": {
        ...
        "files": [
            ...
            "vendor-overrides/elastic-adapter/Search/SearchRequest.php"
        ],
        "exclude-from-classmap": [
            "vendor/babenkoivan/elastic-adapter/src/Search/SearchRequest.php"
        ]
    }
...
```

vendor-overrides/elastic-adapter/Search/SearchRequest.php
```php
<?php declare(strict_types=1);

namespace ElasticAdapter\Search;

use Illuminate\Contracts\Support\Arrayable;

final class SearchRequest implements Arrayable {

    /**
     * @var array
     */
    private $request;

    public function __construct(array $query) {
        $this->request['query'] = $query;
    }

    public function highlight(array $highlight): self {
        $this->request['highlight'] = $highlight;

        return $this;
    }

    public function sort(array $sort): self {
        $this->request['sort'] = $sort;

        return $this;
    }

    public function rescore(array $rescore): self {
        $this->request['rescore'] = $rescore;

        return $this;
    }

    public function from(int $from): self {
        $this->request['from'] = $from;

        return $this;
    }

    public function size(int $size): self {
        $this->request['size'] = $size;

        return $this;
    }

    public function suggest(array $suggest): self {
        $this->request['suggest'] = $suggest;

        return $this;
    }

    /**
     * @param bool|string|array $source
     */
    public function source($source): self {
        $this->request['_source'] = $source;

        return $this;
    }

    public function collapse(array $collapse): self {
        $this->request['collapse'] = $collapse;

        return $this;
    }

    public function aggregations(array $aggregations): self {
        $this->request['aggregations'] = $aggregations;

        return $this;
    }

    public function postFilter(array $postFilter): self {
        $this->request['post_filter'] = $postFilter;

        return $this;
    }

    /**
     * @param int|bool $trackTotalHits
     */
    public function trackTotalHits($trackTotalHits): self {
        $this->request['track_total_hits'] = $trackTotalHits;

        return $this;
    }

    public function indicesBoost(array $indicesBoost): self {
        $this->request['indices_boost'] = $indicesBoost;

        return $this;
    }

    public function trackScores(bool $trackScores): self {
        $this->request['track_scores'] = $trackScores;

        return $this;
    }

    public function minScore(float $minScore): self {
        $this->request['min_score'] = $minScore;

        return $this;
    }

    public function scriptFields(array $scriptFields): self {
        $this->request['script_fields'] = $scriptFields;

        return $this;
    }

    public function toArray(): array {
        return array_filter($this->request);
    }

}
```

Specifically I would like to point the attention to the `toArray` function. I do an `array_filter` before returning the `$this->request` so that the empty array would be removed so that the SearchRequest only contains valid structure for elastic search.

It would be really great if this library could support my use case of skipping the query part of SearchRequest.

**Comment by babenkoivan** [comment]
Hi @davisriska, I've just released [elastic-scout-driver-plus v3.3.0](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v3.3.0) with the requested change. You can now use aggregations without query as follows:

```php
$searchResult = Book::searchQuery()
    ->aggregate('max_price', [
            'max' => [
                'field' => 'price',
            ],
	])
    ->execute();
```

**Comment by davisriska** [comment]
Thank you so much @babenkoivan! I really appreciate your work!
